### ISSUE TITLE
How can I show the stems?

### ISSUE THREAD

**Issue by justanotherfoundry** [Mark2Mark/show-stems-pro-plugin#2]
Looks very promising! I’m a bit confused, though. How can I actually show the stems? Do I need to define them like hints? What do the numbers mean?

<img width="618" alt="Screenshot 2021-09-30 at 21 44 00" src="https://user-images.githubusercontent.com/1331354/135520698-3bc2a950-d139-4c43-a43a-170846ed8cbb.png">

**Comment by Mark2Mark** [comment]
You need to have them in the dimensions palette. That’s where it looks for.

I am actually also thinking to add a way to show the stem values, but I had no requests for that yet.

Does that help?

**Comment by justanotherfoundry** [comment]
Thanks, I just entered the dimensions on the dimensions palette. Maybe that hint could be placed a bit more prominently? It wasn’t obvious to me.

Now I am getting the stem indicators only in elements that lie under the baseline, i.e. descenders. None are shown in the n, for example. Anything else I am doing wrong?

What do the numbers shown by the plug-in (i.e. 2484, 77, 274, 1024 and 250) mean?

<img width="897" alt="Screenshot 2021-10-01 at 21 43 40" src="https://user-images.githubusercontent.com/1331354/135678282-3a4274f2-da4c-45d1-a358-210a4027cd8d.png">

**Comment by justanotherfoundry** [comment]
Also: Is there any support for showing the serif thickness?

**Comment by Mark2Mark** [comment]
> Thanks, I just entered the dimensions on the dimensions palette. Maybe that hint could be placed a bit more prominently? It wasn’t obvious to me.

Good point, thanks. I will highlight that better in the documentations.

Currently you still don’t see the "almost" matches, because I have a threshold at 10 units. e.g. your /n has a stem of 90 in the dimensions palette, but your current /n drawing has 77. If you change the drawing or the number in the dimensions palette, you should see the effect.

> Now I am getting the stem indicators only in elements that lie under the baseline, i.e. descenders. None are shown in the n, for example. Anything else I am doing wrong?

There’s something else that you might not know yet: you can move the measurement beams with the cursor and the key combinations:
_Move vertical beam:_
<kbd>opt+shift</kbd> +drag mouse

_Move horizontal beam:_
<kbd>opt</kbd> +drag mouse

Those are basically like measurement lines, but they stay where you put them. Following into your next question:

> What do the numbers shown by the plug-in (i.e. 2484, 77, 274, 1024 and 250) mean?

These are the distances on each of the measurement lines (horizontal and vertical).

> Also: Is there any support for showing the serif thickness?
Sure, it should be the same, you just need to move the measurement beams to the serifs. Or did I misunderstand?

**Comment by Mark2Mark** [comment]
Here’s a little video of it in use together with your RMX Tuner:

https://user-images.githubusercontent.com/5824595/135741202-8dd54d3e-37ea-44ac-b7a7-efe4f4514fff.mov

**Comment by justanotherfoundry** [comment]
> So long, here is my ["documentation" that I had](https://markfromberg.com/projects/stems/).

Oh, now I understand. I had no idea that page exists.

> Currently you still don’t see the "almost" matches, because I have a threshold at 10 units. e.g. your /n has a stem of 90 in the dimensions palette, but your current /n drawing has 77. 

No, my n has a stem thickness of 90, as you can see from the screenshot (the width of the selection is 90).

> There’s something else that you might not know yet: you can move the measurement beams with the cursor and the key combinations: _Move vertical beam:_ opt+shift +drag mouse

So, that means I need to drag the position of these lines when I move to a different glyph? I thought this was a tool that allows me to quickly step through the glyphs in my font and check whether the stems are right.

> > Also: Is there any support for showing the serif thickness?
> > Sure, it should be the same, you just need to move the measurement beams to the serifs. Or did I misunderstand?

I was assuming the main purpose of this tool is to conveniently and quickly check whether all the stems (and possibly serifs) have the desired thickness. I can’t enter serif thicknesses in the dimensions palette, though. Can you explain what this tool does that the built-in tools in Glyphs don’t offer?
Thanks!

**Comment by Mark2Mark** [comment]
> No, my n has a stem thickness of 90, as you can see from the screenshot (the width of the selection is 90).

Sorry my bad, but in referred to where the measurement line is in the image.

> I was assuming the main purpose of this tool is to conveniently and quickly check whether all the stems (and possibly serifs) have the desired thickness.

Well, it kind of does. Sometimes you need to readjust the position a bit, but it’s still more convenient than measuring manually. The more automated detection is on the list.

> I can’t enter serif thicknesses in the dimensions palette, though.

If you need that, ask Georg if that could be plausible to add to the dimensions palette.

> Can you explain what this tool does that the built-in tools in Glyphs don’t offer?

In some scripts it is more useful than for Lati (Hangeul, Khmer, Lao, …). If you prefer the built-in tools, go for those. No one forces you to use this one. But in my experience, and I use it since long, it is way more convenient to measure things in real time while working on the paths instead of activating the measurement tool and dragging a measurement all the time you need to know values. On the other hand you surely can clutter your file with guidelines all over, click to create them, click to activate that they shall show the measurements, and repeat for each master and each glyph. And maybe then clean them up again. Both these ways also does’t tell you if you have a match or are close to it (which at the moment only works with the build in dimensions palette, but will be extended to read from the font info stems as well.

**Comment by justanotherfoundry** [comment]
Thanks for the explanations!
